src/main/java/com/redhat/rhjmc/containerjfr/net/HttpServer.java

```diff
@@ -19,18 +24,92 @@
     private final Vertx vertx;
     private final HandlerDelegate<HttpServerRequest> requestHandlerDelegate = new HandlerDelegate<>();
     private final HandlerDelegate<ServerWebSocket> websocketHandlerDelegate = new HandlerDelegate<>();
+    private final boolean sslEnabled;
 
     private final io.vertx.core.http.HttpServer server;
-    
 
-    HttpServer(NetworkConfiguration netConf, Logger logger) {
+    private static final String KEYSTORE_PATH_ENV = "KEYSTORE_PATH";
+    private static final String KEYSTORE_PASS_ENV = "KEYSTORE_PASS";
+    private static final String KEY_PATH_ENV = "KEY_PATH";
+    private static final String CERT_PATH_ENV = "CERT_PATH";
+
+    HttpServer(NetworkConfiguration netConf, Environment env, Logger logger) {
         this.netConf = netConf;
         this.logger = logger;
         this.vertx = Vertx.vertx();
-        this.server = vertx.createHttpServer(new HttpServerOptions()
+
+        HttpServerOptions options = new HttpServerOptions()
                 .setCompressionSupported(true)
-                .setLogActivity(true)
-        );
+                .setLogActivity(true);
+        this.sslEnabled = setSslOptions(env, options);
+
+        this.server = vertx.createHttpServer(options);
+    }
+
+    private boolean setSslOptions(Environment env, HttpServerOptions options) {
+        String keystorePath = null;
+        if (env.hasEnv(KEYSTORE_PATH_ENV)) {
+            keystorePath = env.getEnv(KEYSTORE_PATH_ENV);
+        } else if (new File(System.getProperty("user.home"), "container-jfr-keystore.jks").exists()) {
+            keystorePath = System.getProperty("user.home") + File.separator + "container-jfr-keystore.jks";
+        } else if (new File(System.getProperty("user.home"), "container-jfr-keystore.pfx").exists()) {
+            keystorePath = System.getProperty("user.home") + File.separator + "container-jfr-keystore.pfx";
+        } else if (new File(System.getProperty("user.home"), "container-jfr-keystore.p12").exists()) {
+            keystorePath = System.getProperty("user.home") + File.separator + "container-jfr-keystore.p12";
```

Comment:
The keystorePath should probably be an actual `Path` type rather than a String, and methods like `Path.resolve()` used to construct the path rather than appending separators and filenames.